apps/passport-server/src/services/telegramService.ts

```diff
@@ -27,21 +31,33 @@ import { logger } from "../util/logger";
 import { isLocalServer, sleep } from "../util/util";
 import { RollbarService } from "./rollbarService";
 
-const SRW_EVENT_ID_STAGING = "3fa6164c-4785-11ee-8178-763dbf30819c";
-const SRW_EVENT_ID_PROD = "264b2536-479c-11ee-8153-de1f187f7393";
-const TICKETING_PUBKEY_STAGING = [
-  "7cf4d97878d663502339c2baae74b12dcdd229279a9f6bfc83b167e808a32d26",
-  "c5a04b56d0f2d6b1ec10aa1b17298e31b4a087bdabd4a75d9523779e7dca5a17"
-];
-const TICKETING_PUBKEY_PROD = [
-  "a7da882cd090c14a62b70cf07010c1cabb373b17ebd2d120c9de039ceaedfa24",
-  "509e44aa56e97a34e9a54534ef79d484d801757720d18ed872e93dd9de126b09"
+const ALLOWED_EVENT_IDS = [
+  { eventId: "3fa6164c-4785-11ee-8178-763dbf30819c", name: "SRW Staging" },
+  { eventId: "264b2536-479c-11ee-8153-de1f187f7393", name: "SRW Prod" },
+  {
+    eventId: "b03bca82-2d63-11ee-9929-0e084c48e15f",
+    name: "ProgCrypto (Internal Test)"
+  },
+  {
+    eventId: "ae23e4b4-2d63-11ee-9929-0e084c48e15f",
+    name: "AW (Internal Test)"
+  }
 ];
 
+const eventIdIsAllowed = (eventId?: string): boolean => {
+  if (!eventId) throw new Error(`No Event Id found for verification`);
+  return ALLOWED_EVENT_IDS.map((a) => a.eventId).includes(eventId);
+};
+
 export class TelegramService {
   private context: ApplicationContext;
   private bot: Bot;
   private rollbarService: RollbarService | null;
+  // These new variables serve as piece of global state that the dynamic Grammy menu can access
+  // I haven't found a cleaner way to do this yet.
+  private proofUrl: string;
+  private events: LinkedPretixTelegramEvent[];
+  private deleteEvents: boolean;
```

Comment:
i don't think we should have these as class state. `this.events` can be replaced with calls to `this.loadEvents()` since we want the most recent state of the DB, `proofUrl` can be a constant outside the class, and `deleteEvents` being a state variable seems very error-prone and I don't totally understand why we need it

Reply:
@rrrliu `proofUrl` includes `returnUrl`, which needs to know the user's telegramId. So that state does need to be set at run time.